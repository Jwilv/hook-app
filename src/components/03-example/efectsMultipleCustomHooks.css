body{
    padding: 70px;
}
input{
    margin: 25px;
}
p{
    margin: 7px;
}