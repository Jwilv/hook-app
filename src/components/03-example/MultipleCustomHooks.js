import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import { useFetch } from '../../hooks/useFetch'
import './efectsMultipleCustomHooks.css'


export const MultipleCustomHooks = () => {

    const {value,increment} = useCounter(1)

    const { loading,data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${value}`)

    const {author,quote} = !!data && data[0]

    return (
        <>
            <h1>Custom Hooks!!!</h1>
            <hr />

            {
                loading
                    ?
                    (<div className='alert alert-info text-center'> loading...</div>)
                    :
                    (<>
                    <blockquote className='blockquote text-right'>
                        <p className='mb-0'>{quote}</p>
                        <footer className='blockquote'>{author}</footer>
                    </blockquote>
                    <button onClick={increment} className='btn btn-primary'>siguiente frase</button>
                    </>
                    )
            }

        </>
    )
}
