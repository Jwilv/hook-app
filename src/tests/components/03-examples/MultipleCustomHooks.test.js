import React from 'react';
import { shallow } from 'enzyme'
import { MultipleCustomHooks } from '../../../components/03-example/MultipleCustomHooks';



describe('pruebas en <MultipleCustomHooks /> ', () => { 

    test('debe de renderizarse correctamente ', () => {
        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot();
    });
 })