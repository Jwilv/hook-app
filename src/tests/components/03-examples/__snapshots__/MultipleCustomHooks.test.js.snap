// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <MultipleCustomHooks />  debe de renderizarse correctamente  1`] = `
<Fragment>
  <h1>
    Custom Hooks!!!
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
     loading...
  </div>
</Fragment>
`;
